lib/elixir/lib/io.ex

```diff
@@ -252,13 +257,14 @@ defmodule IO do
   @doc """
   Inspects `item` according to the given options using the IO `device`.
 
-  See `Inspect.Opts` for a full list of options.
+  See `inspect/2` for a full list of options.
   """
   @spec inspect(device, item, Keyword.t) :: item when item: var
   def inspect(device, item, opts) when is_list(opts) do
+    tag    = if (tag = opts[:tag]), do: [to_string(tag), ": "], else: ""
```

Comment:
I think we should require the tag to be a binary (in order not to end up in weird situations). Either we just use `[tag, ": "]` or we use `if (tag = opts[:tag]), do: tag <> ": ", else: ""` which will fail if `tag` is not a binary. Wdyt?